### PR TITLE
ipn/ipnlocal: simplify suggest exit node tests

### PR DESCRIPTION
**Pull request by sailorfrag**
This mostly removes a lot of repetition by predefining some nodes and other data structures, plus adds some helpers for creating Peer entries in the netmap. Several existing test cases were reworked to ensure better coverage of edge cases, and several new test cases were added to handle some additional responsibility that is in (or will be shortly moving in) suggestExitNode().

Updates tailscale/corp#19681

Change-Id: Ie14c2988d7fd482f7d6a877f78525f7788669b85

### CODE REVIEW

File: ipn/ipnlocal/local_test.go [hunk 1]
```diff
@@ -3247,104 +3200,57 @@ func TestSuggestExitNode(t *testing.T) {
...
-				t.Errorf("got value %v want %v", got, tt.wantValue)
-				if tt.wantValid != got.Valid() {
-					t.Errorf("got invalid candidate expected valid")
-				}
-				if tt.wantValid {
-					if !reflect.DeepEqual(got, tt.wantValue) {
-						t.Errorf("got value %v want %v", got, tt.wantValue)
-					}
-				}
+			gotID := tailcfg.StableNodeID("<invalid>")
```
Comment (sailorfrag) [inline ipn/ipnlocal/local_test.go:3248]: This is kind of awkward but I have a follow on PR that's going to make `pickWeighted` return a slice in case there's a tie, and this approach makes more sense then.

**Review by agottardo — APPROVED** [review]
